### PR TITLE
Read and write 16-bit PNG files

### PR DESCRIPTION
**Pull request by y-guyon**
Allow and check RGB-YUV conversion for 16-bit samples.

Note: This PR is necessary for https://github.com/AOMediaCodec/libavif/pull/1215.

### CODE REVIEW

**Comment by jzern** [comment]
Maybe a changelog entry?

**Review by jzern — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ avifEncoder struct.
   maxQuantizer are left initialized to the default values. Code should be
   updated to set the quality member. Similarly for the alpha quantizers and
   qualityAlpha.
+* avifImageRGBToYUV() and avifImageYUVToRGB() handle bit depths 8, 10, 12 and
```
Comment (wantehchang) [inline CHANGELOG.md:42]: Nit: Add "avifImage" before "bit depths" because "avifRGBImage" also has a bit depth.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -113,6 +113,16 @@ if(AVIF_ENABLE_GTEST)
     target_include_directories(avifopaquetest PRIVATE ${GTEST_INCLUDE_DIRS})
     add_test(NAME avifopaquetest COMMAND avifopaquetest)
 
+    add_executable(avifpng16btest gtest/avifpng16btest.cc)
+    target_link_libraries(avifpng16btest aviftest_helpers ${GTEST_LIBRARIES})
+    target_include_directories(avifpng16btest PRIVATE ${GTEST_INCLUDE_DIRS})
```
Comment (wantehchang) [inline tests/CMakeLists.txt:118]: Nit: I suggest changing "16b" to "16bit" or "16bpc" in the test/file names and the test image name.

**Review by wantehchang — APPROVED** [review]
LGTM. I suggest two changes.

**Pull request merged**